rossum_api/models/hook.py

```diff
@@ -84,21 +105,116 @@ class Hook:
-    events: list[str] = field(default_factory=list)
-    settings: dict[str, Any] = field(default_factory=dict)
-    settings_schema: dict[str, Any] | None = None
-    secrets: dict[str, Any] = field(default_factory=dict)
-    extension_source: str = "custom"
+    events: list[HookEvent] = field(default_factory=list)
+    settings: JsonDict = field(default_factory=dict)
+    settings_schema: JsonDict | None = None
+    secrets: JsonDict = field(default_factory=dict)
+    extension_source: HookExtensionSource = "custom"
     sideload: list[str] = field(default_factory=list)
     token_owner: str | None = None
     token_lifetime_s: int | None = None
     description: str | None = None
+
+
+@dataclass
+class HookRunData:
+    """Data class for hook execution logs.
+
+    HookRunData captures detailed execution logs and metadata for webhook/hook runs within the system.
+    It provides structured logging for tracking hook lifecycle events, performance metrics, and debugging information.
+
+    Notes
+    -----
+    The ``timestamp`` field is automatically set to the current UTC time if not provided.
+
+    The ``start`` and ``end`` fields should be used to track execution duration for performance monitoring.
+
+    Attributes
+    ----------
+    log_level
+        The severity level of the log entry: "INFO" for successful execution, "ERROR" for failures,
+        or "WARNING" for non-critical issues.
+    action
+        The specific action being performed by the hook.
+    event
+        The event type that triggered the hook execution.
+    request_id
+        Unique identifier for the HTTP request that initiated this hook execution.
+    organization_id
+        The ID of the organization that owns the hook configuration.
+    hook_id
+        The unique identifier of the hook configuration being executed.
+    hook_type
+        The type/category of the hook (e.g., webhook, email, custom integration).
+    queue_id
+        Reference to the queue where the document/annotation is located.
+    annotation_id
+        Reference to the specific annotation that triggered or is associated with the hook execution.
+    email_id
+        Reference to an email record if the hook involves email processing.
+    message
+        Human-readable description or log message providing additional context about the execution.
+    request
+        Serialized representation of the outgoing HTTP request body sent by the hook (typically JSON).
+    response
+        Serialized representation of the HTTP response received from the hook endpoint.
+    start
+        ISO 8601 timestamp indicating when the hook execution started.
+    end
+        ISO 8601 timestamp indicating when the hook execution completed.
+    settings
+        Dictionary containing hook-specific configuration settings and parameters used during execution.
+    status
+        Text description of the execution status (e.g., "success", "failed", "timeout").
```

Comment:
I think we have the status currently defined like:

```python
class HookRunStatus(models.TextChoices):
    waiting = auto()
    running = auto()
    completed = auto()
    failed = auto()
    cancelled = auto()
```

but it can still change as the api is beta and changes are expected.